core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -4721,24 +4551,132 @@ public DisplayType getDisplayType() {
     return mDisplayType;
   }
 
+  private static final Map<PropertyKey, Deprecated> DEPRECATED_KEYS =
+      populateAnnotatedKeyMap(PropertyKey.class, PropertyKey.class, Deprecated.class);
+  private static final Map<Template, Deprecated> DEPRECATED_TEMPLATES =
+      populateAnnotatedKeyMap(Template.class, Template.class, Deprecated.class);
+
+  private static final Map<PropertyKey, Removed> REMOVED_KEYS =
+      populateAnnotatedKeyMap(RemovedKey.class, PropertyKey.class, Removed.class);
+  private static final Map<Template, Removed> REMOVED_TEMPLATES =
+      populateAnnotatedKeyMap(Template.class, Template.class, Removed.class);
+
   /**
-   * @param name the name of a property key
+   * Given a class to search, a field type, and an annotation type will return a map of all
+   * fields which are marked with the given annotation to the instance of the annotation.
+   *
+   * @param searchClass the class to search through for fields
+   * @param fieldType the class of the field to search for
+   * @param annotationClazz the annotation to look for
+   * @param <I> any class
+   * @param <J> any class
+   * @param <K> a class extending Annotation
+   * @return a map
+   */
+  private static <I, J, K extends Annotation> Map<J, K> populateAnnotatedKeyMap(
```

Comment:
Can we make this more object-oriented, e.g. by changing this method to be a constructor for a class, and having `getKeyAnnotation` be a method on that class?

Reply:
Updated. See 331d111  with `AnnotatedKeyChecker`